src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const resObj: GroupsMap<T> = {};
+
+  items.forEach((item: T) => {
```

Comment:
You can use reduce instead forEach. In this case you don't need to create resObj on 6 line.